### DISCUSSION TITLE
Adding collector to image

### DISCUSSION THREAD

**Discussion by acraftydev** [aws-observability/aws-otel-community#681] [Q&A]
I am using the AWS collector sidecar on ecs, i want to collect metrics from my apache server, using the available apache receiver.
https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/apachereceiver

How can i add a reciever to my image, is there an example of customising the image and including modules from the contrib project?

thanks

**Comment by mhausenblas** [comment]
Thanks for reaching out @acraftydev!

> How can i add a reciever to my image, is there an example of customising the image and including modules from the contrib project?

Other than using ADOT as your upstream (see also [dev](https://github.com/aws-observability/aws-otel-collector/blob/main/docs/developers/README.md)) there is no way. Did you consider building a [custom distribution](https://opentelemetry.io/docs/collector/distributions/#custom-distributions)?
